Sources/ProjectDescription/Workspace.swift

```diff
@@ -43,7 +60,7 @@ public struct Workspace: Codable, Equatable {
     public let additionalFiles: [FileElement]
 
     /// Generation options.
-    public let generationOptions: GenerationOptions?
+    public let generationOptions: [GenerationOption]
```

Comment:
Having the struct was more correct as an API 😞 we wanted to change `ProjectOption` to follow suit however needed to wait for 3.0 to do so - see the original rationale in 

https://github.com/tuist/tuist/pull/3832#discussion_r769845058

Reply:
If the main "selling point" of the struct approach was to avoid custom codables, it is now automatically solved because  they are now automatically synthesized by the swift compiler.
Do you still think we should go with struct instead of enum?

In any case, I would do it in a separate PR, to address all the various options enums/struct